lte/gateway/c/core/oai/tasks/mme_app/mme_app_state_manager.cpp

```diff
@@ -219,11 +219,10 @@ status_code_e MmeNasStateManager::read_ue_state_from_db() {
     for (const auto& key : keys) {
       OAILOG_DEBUG(log_task, "Reading UE state from db for %s", key.c_str());
       oai::UeContext ue_proto = oai::UeContext();
+      auto* ue_context = (ue_mm_context_t*)(calloc(1, sizeof(ue_mm_context_t)));
```

Comment:
Please use the interpret_cast here.

Reply:
Hi @ajahl, what is the interpretation here? This change was an unrelated refactoring to the (reverted) MME_BENCHMARK feature?

Changed back to the `reinterpret_cast` state (i.e., file is removed from change set).

Reply:
> what is the interpretation here?

`Reinterpret_cast` is the equivalent in C++ to C standard cast that is used in line 222. It is recommended by the C++ standard. And in C++ there are more specific casts, for example, `const_cast`, `static_cast`, `dynamic_cast` and so on. You don’t know which cast the C++ compiler will use when you apply the C standard cast, which may lead to an undefined behaviour. And there are more differences. The C standard cast breaks constness. In C++ you must explicitly use the const_cast for that. And in contract to C standard cast, you get an understandable error output.